rye/src/lock.rs

```diff
@@ -136,6 +143,7 @@ pub fn update_workspace_lockfile(
     output: CommandOutput,
     sources: &ExpandedSources,
     lock_options: &LockOptions,
+    keyring_provider: KeyringProvider,
```

Comment:
Should this be on `LockOptions`, rather than passed in as a standalone argument?

Reply:
(I decided: no, because it's also needed for `sync` -- which does take `LockOptions`, but for unrelated reasons.)